### ISSUE TITLE
Selected edges do not become bold

### ISSUE THREAD

**Issue by aaronst** [almende/vis#2990]
When using the `dot` shape for nodes, selected edges' labels do not become bold.  See here for an example: https://jsfiddle.net/vyh3x23n/.

**Comment by rasmusblockzero** [comment]
In the Fiddle the edge does become bold both on select and hover. (?)

**Comment by rasmusblockzero** [comment]
The example is using the chosen function.

I would rephrase the issue as "chosen function doesn't work when set on network.edges but when set on an individual edge".

Or at least that's what my first check reveals.

@eymiha or @mojoaxel this seems to have been introduced (or refactored end of last year becf18e6ebfc1abcf650a429b3c75500665caa4a ) Would any of you have any ideas?

**Comment by wimrijnders** [comment]
Found it. Unlike the drawing code for node labels, the code from edges labels was missing a conditional recalculation of font attributes due to selection and hover.

In pseudo code, which is perhaps easier to understand:
```
if select or hover state changed
  redetermine the label font
end
draw the label
```
-----
What will happen now, is that the label will turn bold if:
- The cursor hovers above it
- The edge is selected
- The cursor hovers above one of the nodes connected to the edge
- A node connected to the edge is selected

This is exactly how the rendering of the edge curve works. If any of these does not hold, the label will return to normal font weight.

I hope this is according to expectations; please advise if it is not.

**Comment by aaronst** [comment]
@wimrijnders To clarify, the last scenario could be disabled by specifying `interaction.selectConnectedEdges = false`, correct?

**Comment by wimrijnders** [comment]
Good one! Label boldness is not disabled. Will fix it ASAP.

Actually, I'll need to go through all options. Thanks for the tip!

**Comment by rasmusblockzero** [comment]
edge.labelHighlightBold would condition this in total I believe
and the edge.chosen option would be specifically juggly as it also can have the sub feature edge.chosen.label 
:-)

**Comment by wimrijnders** [comment]
@rasmusblockzero Aargh! Need to look at those as well.
